src/control/dmg/pool.go

```diff
@@ -23,7 +23,177 @@
+	pb "github.com/daos-stack/daos/src/control/common/proto/mgmt"
+	"github.com/inhies/go-bytesize"
+	"github.com/pkg/errors"
+)
+
+const (
+	msgSizeNoNumber = "size string doesn't specify a number"
+	msgSizeBad      = "size string illegal, %s"
+	msgSizeZeroScm  = "non-zero scm size is required"
+	maxNumSvcReps   = 13
+)
 
 // PoolCmd is the struct representing the top-level pool subcommand.
-type PoolCmd struct{}
+type PoolCmd struct {
+	Create CreatePoolCmd `command:"create" alias:"c" description:"Create a DAOS pool"`
+}
+
+// CreatePoolCmd is the struct representing the command to create a DAOS pool.
+type CreatePoolCmd struct {
+	GroupName  string `short:"g" long:"group" description:"DAOS pool to be owned by given group"`
+	UserName   string `short:"u" long:"user" description:"DAOS pool to be owned by given user"`
+	ACLFile    string `short:"a" long:"acl-file" description:"Access Control List file path for DAOS pool"`
+	ScmSize    string `short:"s" long:"scm-size" required:"1" description:"Size of SCM component of DAOS pool"`
+	NVMeSize   string `short:"n" long:"nvme-size" description:"Size of NVMe component of DAOS pool"`
+	RankList   string `short:"r" long:"ranks" description:"DAOS pool to be owned by given user"`
+	NumSvcReps uint32 `short:"v" long:"nsvc" default:"1" description:"Number of pool service replicas"`
+	//GID uint32 `short:"g" long:"gid" description:"Pool GID"`
+	//UID uint32 `short:"u" long:"uid" description:"Pool UID"`
+	//Group string `short:"G" long:"group" description:"Pool group name"`
+	//Mode uint64 `short:"m" long:"mode" description:"Pool permissions mode"`
+}
+
+// getSize retrieves number of bytes from human readable string representation
+func getSize(sizeStr string) (bytesize.ByteSize, error) {
+	if sizeStr == "" {
+		return bytesize.New(0.00), nil
+	}
+	if common.IsAlphabetic(sizeStr) {
+		return bytesize.New(0.00), errors.New(msgSizeNoNumber)
+	}
+
+	// change any alphabetic characters to upper before ByteSize.parse()
+	sizeStr = strings.ToUpper(sizeStr)
+
+	// append "B" character if absent (required by ByteSize.parse())
+	if !strings.HasSuffix(sizeStr, "B") {
+		sizeStr += "B"
+	}
+
+	return bytesize.Parse(sizeStr)
+}
+
+// calcStorage calculates SCM & NVMe size for pool from user supplied parameters
+func calcStorage(scmSize string, nvmeSize string) (
+	scmBytes bytesize.ByteSize, nvmeBytes bytesize.ByteSize, err error) {
+
+	scmBytes, err = getSize(scmSize)
+	if err != nil {
+		err = errors.WithMessagef(
+			err, "illegal scm size: %s", scmSize)
+		return
+	}
+
+	if scmBytes == 0 {
+		err = errors.New(msgSizeZeroScm)
+		return
+	}
+
+	nvmeBytes, err = getSize(nvmeSize)
+	if err != nil {
+		err = errors.WithMessagef(
+			err, "illegal nvme size: %s", nvmeSize)
+		return
+	}
+
+	ratio := 1.00
+	if nvmeBytes > 0 {
+		ratio = float64(scmBytes) / float64(nvmeBytes)
+	}
+
+	if ratio < 0.01 {
+		fmt.Printf(
+			"SCM:NVMe ratio is less than 1%%, DAOS performance " +
+				"will suffer!\n")
```

Comment:
Would it make sense to make this an interactive "are you sure?" type prompt?

Reply:
@johannlombardi as you requested this warning, would you like this to prompt for confirmation?